File: solutions/237 Delete Node in a Linked List [Medium]/sol.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node){
        *node=*node->next;
    }
};

// C++
// 0 ms (beats 100.00%)
// 11.32 mb (beats 42.93%)

File: solutions/23 Merge k Sorted Lists [Hard]/sol.py
class Solution:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        if not lists or len(lists) == 0: return None

        while len(lists) > 1:
            t = []
            for i in range(0, len(lists), 2):
                a = lists[i]; b = lists[i+1] if i + 1 < len(lists) else None
                t.append(self.mergeTwoLists(a, b))
            lists = t

        return lists[0]
    
    def mergeTwoLists(self, a, b):
        node = ListNode(); ans = node

        while a and b:
            if a.val > b.val:
                node.next = b; b = b.next
            else:
                node.next = a; a = a.next
            
            node = node.next
        
        if a:
            node.next = a
        if b:
            node.next = b

        return ans.next
    
# Python    
# 77 ms (beats 63.88%)
# 19.28 mb (beats 86.03%)